client/app/components/group/_storeList/storeList.controller.js

```diff
@@ -17,7 +17,6 @@ class StoreListController {
     if (changes.groupId && angular.isDefined(changes.groupId.currentValue)) {
       this.Store.listByGroupId(changes.groupId.currentValue).then((data) => {
         this.CurrentStores.set(data);
-        this.storeList = angular.copy(this.CurrentStores.list);
```

Comment:
The only reason I could imagine is to keep storeMap reusable, but that's not important at all.
So yes, I would remove the binding. The watch is still necessary, though.